provider/keycloak_openid_client_authorization_permission.go

```diff
@@ -0,0 +1,177 @@
+package provider
+
+import (
+	"fmt"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/terraform/helper/validation"
+	"github.com/mrparkers/terraform-provider-keycloak/keycloak"
+	"strings"
+)
+
+var (
+	keycloakOpenidClientResourcePermissionDecisionStrategies = []string{"UNANIMOUS", "AFFIRMATIVE", "CONSENSUS"}
+	keycloakOpenidClientPermissionTypes                      = []string{"resource", "scope"}
+)
+
+func resourceKeycloakOpenidClientAuthorizationPermission() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceKeycloakOpenidClientAuthorizationPermissionCreate,
+		Read:   resourceKeycloakOpenidClientAuthorizationPermissionRead,
+		Delete: resourceKeycloakOpenidClientAuthorizationPermissionDelete,
+		Update: resourceKeycloakOpenidClientAuthorizationPermissionUpdate,
+		// This resource can be imported using {{realm}}/{{resource_server_id}}. The Client ID is displayed in the GUI
+		Importer: &schema.ResourceImporter{
+			State: resourceKeycloakOpenidClientAuthorizationPermissionImport,
+		},
+		Schema: map[string]*schema.Schema{
+			"resource_server_id": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"realm_id": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"description": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"decision_strategy": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ValidateFunc: validation.StringInSlice(keycloakOpenidClientResourcePermissionDecisionStrategies, false),
+				Default:      "UNANIMOUS",
+			},
+			"policies": {
+				Type:     schema.TypeList,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+				Optional: true,
+			},
+			"resources": {
+				Type:     schema.TypeList,
```

Comment:
Does the order of `resources` matter?  If not, this should be a `TypeSet`.

Reply:
Done